projects/step-core/src/lib/modules/table/services/items-per-page.service.ts

```diff
@@ -0,0 +1,34 @@
+import { Injectable } from '@angular/core';
+import { UserService } from '../../../client/generated';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ItemsPerPageService {
+  private defaultItemsPerPage: Array<number> = [10, 25, 50, 100];
+  private customItemsPerPage?: number | boolean;
+
+  constructor(private _userService: UserService) {}
+
+  public getItemsPerPage(loadedUserPreferences: Function): any {
```

Comment:
I think the type ```Function``` is too abstract, it would be better to define it as ```(itemsPerPage: number) => void``` imo

Reply:
also the return type seems to be ```number``` at a first glance, perhaps its ```any``` for a reason?

Reply:
The return type is actually Array<number>, but good point.